### ISSUE TITLE
TypeScript definitions do not match implementation

### ISSUE THREAD

**Issue by damianb** [nicoespeon/gitgraph.js#179]
Noticing a number of issues when trying to use the current Typescript definitions.

``` js
let develop = master.branch({
  name: 'develop',
  column: 1
})
```
results in:
```
error TS2345: Argument of type '{ name: string; column: number; }' is not assignable to parameter of type 'string | BranchOptions'.
  Object literal may only specify known properties, and 'column' does not exist in type 'string | BranchOptions'.
```
(seems that BranchOptions does not specify column as a valid option...)

---

``` js
storyBranch2
  .commit('S-00004: Interim commit from another feature branch')
  .merge(develop)
```
results in:
```
error TS2339: Property 'merge' does not exist on type 'void'.
```
(implying that `branch.commit()` returns `void`? which it does not seem to; it seems to be a fluent interface.)

---

``` js
let gitTheme = new GitGraph.Template({
  colors: [ '#cc9', '#9c9', '#99f', '#c9c', '#c99', '#9cc' ], // branches colors, 1 per column
  branch: {
    color: '#333',
    lineWidth: 1,
    spacingX: 50,
    mergeStyle: 'straight',
    labelFont: 'normal 12px Roboto',
    showLabel: true
  }
})
```
results in: 
```
error TS2345: Argument of type '{ colors: string[]; branch: { color: string; lineWidth: number; spacingX: number; mergeStyle: str...' is not assignable to parameter of type 'TemplateOptions'.
  Types of property 'branch' are incompatible.
    Type '{ color: string; lineWidth: number; spacingX: number; mergeStyle: string; labelFont: string; show...' is not assignable to type '{ color?: string | undefined; lineWidth?: number | undefined; lineDash?: number[] | undefined; me...'.
      Object literal may only specify known properties, and 'labelFont' does not exist in type '{ color?: string | undefined; lineWidth?: number | undefined; lineDash?: number[] | undefined; me...'.
```
(seems that labelFont is available on the Commit, but was forgotten in the Template options for branches? Maybe it needs to be branch level too?)

---

``` js
let master = graph.branch({
  name: 'master',
  color: '#a00',
  commitDefaultOptions: {
    dotColor: '#f77'
  },
  column: columns.master
})
```
results in:
```
error TS2345: Argument of type '{ name: string; color: string; commitDefaultOptions: { dotColor: string; }; column: number; }' is not assignable to parameter of type 'string | BranchOptions'.
  Object literal may only specify known properties, and 'color' does not exist in type 'string | BranchOptions'.
```
(looks like branchoptions doesn't accept color, yet specifying color on the branch level seems to have a noticeable effect on the branch line color...)

---

Also noticing that many of these are also undocumented in the JSdocs, too.

**Comment by fabien0102** [comment]
😕 It's an old project with also an old architecture (only one file…) So all the typescript definition are generated from JSdoc with a custom script ^^. No JSdoc, no typescript definition…

If you can complete the JSdoc to fix this and make a PR, it should be awesome 😄 

Note: You can generate types with `npm run tsd` to test the generated definition 😉

**Comment by nicoespeon** [comment]
IMO, this issue will be fixed with #187 since we use proper TypeScript instead of creating types from the JS API 👌

**Comment by nicoespeon** [comment]
So, this is technically _not_ resolved for the `gitgraph.js` package… But we released `@gitgraph/react` in v1.0.0-0 and this issue was fixed here—it's a TS project, so TS definitions do match.

Thus, I'm closing this issue as we won't maintain the `gitgraph.js` package. There will be a `@gitgraph/js` package soon to replace it. 

We still accept PRs if someone really needs it, but I'd suggest moving to the aforementioned packages. Thanks 🙏
